### PR TITLE
Update configuration flags for ISA-JSON compliance

### LINKED ISSUE #1602: DataHub Configuration - add Template to SP advanced
The feature in the admin panel "Sample type Template enabled" (Whether users can create and manage Templates for Sample types) should be part of "Advanced features enabled" (If set, the advance features of the single page is enabled.)

Instance admin should not be able to enable Template without SinglePage advanced.

### LINKED ISSUE #1604: "Query for Template" button visible for all assets, not only Samples
**Describe the bug**
When browsing any asset or ISA element, the "Query for Template" button that leads to the querying of Samples by Templates is displayed next to the new button.

**To Reproduce**
1. Go to 'Browse' > Investigations
2. See button

**Expected behavior**
The button should only appear on the Samples index page.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by somathias**
Content:

- fix #1602: Sample Type Templates are now part of what was formerly called "SP advanced enabled"
- SP advanced has been renamed "Compliance with ISA-JSON schemas enabled" and moved to under the "SEEK features" section in the admin configuration panel
- fix #1604: The Query by Template button is only displayed for Samples and only if ISA-JSON compliancy is enabled. 

Note: 
- this includes the changes requested for PR #1605 which can therefore be closed
- this PR might cause conflict with PR #1610 and will probably need to be updated depending on which is merged first.

### CODE REVIEW

File: app/views/admin/features_enabled.html.erb [hunk 1]
```diff
@@ -28,6 +28,17 @@
                            'Help URL', "This is the URL for the external help e.g #{Seek::Config.default_external_help_url}.") %>
   </div>
 
+  <%= admin_checkbox_setting(:project_single_page_enabled, 1, Seek::Config.project_single_page_enabled,
+                             "Single page enabled", "The single page UI for #{t(:project).pluralize}", :onchange=>toggle_appear_javascript('single_page_folders')) %>
+  <div id="single_page_folders" class="additional_settings" style="<%= show_or_hide_block Seek::Config.project_single_page_enabled -%>">
+    <%= admin_checkbox_setting(:project_single_page_folders_enabled, 1, Seek::Config.project_single_page_folders_enabled,
+                               "Project folders enabled","If set, the folders of the projects will be shown in the tree view.") %>
+  </div>
+  <%= admin_checkbox_setting(:isa_json_compliance_enabled, 1, Seek::Config.isa_json_compliance_enabled,
```
Comment (kdp-cloud) [inline app/views/admin/features_enabled.html.erb:37]: Something to maybe think about:
Would be nice if the user is only able to select this option if both 'Single Page enabled' and 'ISA enabled' are checked.

File: app/views/samples/index.html.erb [hunk 2]
```diff
@@ -25,5 +25,5 @@
       </script>
   <% end %>
 <% else %>
-    <%= render partial: "general/index" %>
+    <%= render partial: "general/index", :locals=>{:show_samples_query_button=>Seek::Config.isa_json_compliance_enabled}  %>
```
Comment (kdp-cloud) [inline app/views/samples/index.html.erb:28]:
```suggestion
    <%= render partial: "general/index", locals: { show_samples_query_button: Seek::Config.isa_json_compliance_enabled }  %>
```
Just a minor change.

**Review by kdp-cloud — CHANGES_REQUESTED** [review]
Just one suggestion...
Other than, Nice work!

File: app/views/admin/features_enabled.html.erb (hunk 1 above)
Comment (somathias) [reply app/views/admin/features_enabled.html.erb:37]: Good suggestion, we will look into some javascript magic to do this

**Review by kdp-cloud — APPROVED** [review]

**Review by reutenauer — APPROVED** [review]

**Review by stuzart — APPROVED** [review]

**Pull request merged**